collect_app/src/main/java/org/odk/collect/android/widgets/DateTimeWidget.java

```diff
@@ -15,139 +15,163 @@
 package org.odk.collect.android.widgets;
 
 import android.annotation.SuppressLint;
+import android.app.Activity;
 import android.content.Context;
-import android.widget.LinearLayout;
+import android.util.TypedValue;
+import android.view.View;
 
 import org.javarosa.core.model.data.DateTimeData;
 import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.joda.time.DateTime;
 import org.joda.time.LocalDateTime;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.databinding.DateTimeWidgetAnswerBinding;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.listeners.WidgetValueChangedListener;
 import org.odk.collect.android.widgets.interfaces.BinaryDataReceiver;
-import org.odk.collect.android.widgets.interfaces.ButtonClickListener;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.DateTimeUtils;
+import org.odk.collect.android.widgets.utilities.DateTimeWidgetUtils;
 
 /**
  * Displays a DatePicker widget. DateWidget handles leap years and does not allow dates that do not
  * exist.
- *
- * @author Carl Hartung (carlhartung@gmail.com)
- * @author Yaw Anokwa (yanokwa@gmail.com)
  */
-
 @SuppressLint("ViewConstructor")
-public class DateTimeWidget extends QuestionWidget implements BinaryDataReceiver, WidgetValueChangedListener, ButtonClickListener {
+public class DateTimeWidget extends QuestionWidget implements BinaryDataReceiver {
+    DateTimeWidgetAnswerBinding binding;
 
-    DateWidget dateWidget;
-    TimeWidget timeWidget;
+    private LocalDateTime selectedDate;
```

Comment:
Can't we use LocalDateTime here only instead of LocalDateTime and DateTime. DateTime seems redundant since you can save both date and time in LocalDateTime.